crates/code_generation/src/cyclers.rs

```diff
@@ -455,7 +457,16 @@ fn generate_cycle_method(cycler: &Cycler, cyclers: &Cyclers, mode: Execution) ->
     let pre_setup = match mode {
         Execution::None => Default::default(),
         Execution::Run => quote! {
-            let enable_recording = self.enable_recording && self.hardware_interface.should_record();
+            let enable_recording = self.cycles_per_recording != 0 && self.cycles_to_next_recording == 0;
+            if self.cycles_per_recording != 0 {
+                if self.cycles_to_next_recording == 0 {
+                    self.cycles_to_next_recording = self.cycles_per_recording - 1;
+                }
+                else {
+                    self.cycles_to_next_recording -= 1;
+                }
```

Comment:
```suggestion
self.cycles_to_next_recording = (self.cycles_to_next_recording + 1) % self.cycles_per_recording
```
Saves one indentation :)

Reply:
Oh and requires renaming of `cycles_to_next_recording` to sth. like `cycles_since_last_recording`

Reply:
I like the idea, but this would also require changing 

```rust
let enable_recording = self.cycles_per_recording != 0 && self.cycles_to_next_recording == 0;
```

to something like this

```rust
let enable_recording = self.cycles_per_recording != 0 && self.cycles_since_last_recording == self.cycles_per_recording - 1;
```
which might be less readable. Otherwise, the order of assignments might be changed, but that would change the meaning of `cycles_since_last_recording`. 
I'm not sure what to think about these options, what is your opinion about this?

Reply:
Still an idea: extract the counting into its own struct to implement single responsibility principle?

Reply:
I like this

Reply:
I now factored this out to a `RecordingTrigger` struct, which can also be more easily extended in the future.